pkg/imgpkg/cmd/copy.go

```diff
@@ -77,9 +75,12 @@ func (o *CopyOptions) Run() error {
 	logger := ctlimg.NewLogger(os.Stderr)
 	prefixedLogger := logger.NewPrefixedWriter("copy | ")
 
-	registry, err := ctlimg.NewRegistry(o.RegistryFlags.AsRegistryOpts(), imagelayers.NewImageLayerWriterCheck(o.IncludeNonDistributableFlag.IncludeNonDistributable))
+	opts := o.RegistryFlags.AsRegistryOpts()
+	opts.IncludeNonDistributableLayers = o.IncludeNonDistributableFlag.IncludeNonDistributable
```

Comment:
it would be nice if we remove https://github.com/vmware-tanzu/carvel-imgpkg/blob/develop/pkg/imgpkg/cmd/non_distributable_flags.go dedicated struct and just inline field on CopyOptions since it's just a single flag used in one command. this would make `o.IncludeNonDistributableFlag.IncludeNonDistributable` into `o.IncludeNonDistributable`  which i think reads a bit better

Reply:
any thoughts on renaming flag from `--include-non-distributable` to `--include-non-distributable-layers`?

Reply:
My thoughts are:
- this is a slightly better name for this flag.
- We have already shipped with this flag. It would be a breaking change. I would want to capture this change in its own issue / separate PR as that brings concerns that are orthogonal to this 'refactor' PR. 
-- I don't have enough concern for this flag rename to create a GH issue for it. My take is if we get feedback from users that this flag is confusing I would prioritize it accordingly. 

I will go ahead and remove the `IncludeNonDistributableFlag` struct and add it to the `CopyOptions` instead.

Reply:
added https://github.com/vmware-tanzu/carvel-imgpkg/issues/112